core/app/models/workarea/order/items_extension.rb

```diff
@@ -1,12 +1,14 @@
 module Workarea
   class Order
     module ItemsExtension
-      def find_existing(sku, customizations = {})
+      def find_existing(sku, attributes = {})
+        customizations = attributes[:customizations] if attributes.present?
         customizations.stringify_keys! if customizations.present?
 
         detect do |item|
           item.sku == sku &&
-            item.customizations_eql?(customizations)
+            item.customizations_eql?(customizations) &&
```

Comment:
We should include `customizations_eql?` as part of the check, it would be unexpected to check `attributes_eql?` and get back `true` if the customizations were actually different.

Reply:
We don't check `attributes_eql?` until `customizations_eql?` is true. To preserve the functionality from v3.5 and below, we're only going to check the attributes if the SKU and customization check fails.

Reply:
I'm not sure I follow, `attributes_eql?` is really able to replace all the logic in this `detect` block as a point to override and customize that logic without having to get at this awkwardly deportable method.